components/common/pkg/kafka/health/health.go

```diff
@@ -0,0 +1,119 @@
+package health
+
+import (
+	"context"
+	"github.com/kyma-incubator/knative-kafka/components/controller/constants"
+	"go.uber.org/zap"
+	"net/http"
+	"sync"
+)
+
+// Interface For Providing Overrides For Liveness And Readiness Information
+type Status interface {
+	IsAlive() bool
+	IsReady() bool
+}
+
+// Structure Containing Basic Liveness Information For Health Server
+type Server struct {
+	server   *http.Server // The Golang HTTP Server Instance
+	status   Status
+	httpPort string       // The HTTP Port The Dispatcher Server Listens On
+
+	// Synchronization Mutexes
+	liveMutex      sync.Mutex	// Synchronizes access to the liveness flag
+
+	// Internal Flags
+	alive          bool         // A flag that controls the response to liveness requests
+}
+
+// Creates A New Server With Specified Configuration
+func NewHealthServer(httpPort string, healthStatus Status) *Server {
+	health := &Server{
+		httpPort:   httpPort,
+		status:     healthStatus,
+	}
+
+	// Initialize The HTTP Server
+	health.initializeServer(httpPort)
+
+	// Return The Health Server
+	return health
+}
+
+// Synchronized Function To Set Liveness Flag
+func (hs *Server) SetAlive(isAlive bool) {
+	hs.liveMutex.Lock()
+	hs.alive = isAlive
+	hs.liveMutex.Unlock()
+}
+
+// Set All Liveness And Readiness Flags To False
+func (hs *Server) ShuttingDown() {
+	hs.SetAlive(false)
```

Comment:
Method names should be verbs, something like "Shutdown" is better than "ShuttingDown"

Reply:
Altered as suggested